sunpy/net/dataretriever/sources/gong.py

```diff
@@ -0,0 +1,289 @@
+                    ...                 ...    ...                ...
+    2016-06-26 00:00:00 2016-06-27 00:00:00   GONG Data not Available
+    2016-06-27 00:00:00 2016-06-28 00:00:00   GONG Data not Available
+    2016-06-28 00:00:00 2016-06-29 00:00:00   GONG Data not Available
+    2016-06-29 00:00:00 2016-06-30 00:00:00   GONG Data not Available
+    2016-06-30 00:00:00 2016-07-01 00:00:00   GONG Data not Available
+    2016-07-01 00:00:00 2016-07-02 00:00:00   GONG Data not Available
+    2016-07-02 00:00:00 2016-07-03 00:00:00   GONG Data not Available
+    2016-07-03 00:00:00 2016-07-04 00:00:00   GONG Data not Available
+    2016-07-04 00:00:00 2016-07-05 00:00:00   GONG Data not Available
+    2016-07-05 00:00:00 2016-07-06 00:00:00   GONG Data not Available]
+    """
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        returns list of urls corresponding to given TimeRange.
+        """
+        #TO-DO: Figure out where scraper would and wouldn't work
+        table_physobs = {'INTENSITY' : 'i', 'LOS_MAGNETIC_FIELD': 'b'} #Supported physical observations.
+        table_instruments = {'bigbear':'bb', 'cerrotololo':'ct', 'learmonth':'le', 'maunaloa':'ml',
+                             'teide':'td', 'udaipur':'ud', 'tucson':'z'} #For magnetogram and intensity
+
+        physobs_in = ('physobs' in kwargs.keys() and kwargs['physobs'] in table_physobs.keys())#Is PhysObs entered
+        instrument_in = ('instrument' in kwargs.keys() and kwargs['instrument'] in table_instruments.keys())
+        wavelength_in = ('wavelength' in kwargs.keys())
+        #Is instrument entered.
+        url_pattern_1 = 'ftp://gong2.nso.edu/QR/{id}qa/%Y%m/{ObsID}bqa%y%m%d/{ObsID}bqa%y%m%dt%H%M.fits.gz'
+        url_pattern_2 = 'http://gong2.nso.edu/HA/haf/%Y%m/%Y%m%d/%Y%m%d%H%M%S{ObsID}h.fits.fz'
+
+        pattern_table = {6562:url_pattern_1, 6563:url_pattern_1, 6768:url_pattern_2}
+
+        result = list() #final list of urls
+        patterns = list()
+        if not physobs_in:
+            patterns.append(url_pattern_1), patterns.append(url_pattern_2)
+        else:
+            if kwargs['physobs'] == 'LOS_MAGNETIC_FIELD':
+                patterns.append(url_pattern_1)
+            else:
+                #Differentiate on basis of wavelength
+
+                #The isclose function in numpy can only check two arrays if they
+                #are close to each other within some precision. Standalone values such
+                #as int, doubles etc can't be checked that way. In order to do that, we put the
+                #two indiviual values in two seperate arrays da and db and then apply
+                #isclose on both those arrays.
+                if not wavelength_in:
+                    patterns.append(url_pattern_1), patterns.append(url_pattern_2)
+                else:
+                    try:
+                        wave_float = kwargs['wavelength'].min.value
```

Comment:
@wafels: Yes, @dpshelio did suggest we need a function in sunpy to accomadate errors in units. Specifically in ```Wavelength```, but this shouldn't just be limited to Wavelength, it should be available for all units. You have errors in all physical units (and not just wavelength), So this needs a little bit more thought.